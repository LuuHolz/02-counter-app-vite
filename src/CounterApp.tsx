import PropTypes from 'prop-types';
import { useState } from 'react';


const CounterApp = ({ value }: { value: number }) => {
    const [counter, setCounter] = useState( value )

    const handleAdd = () => {
        const suma = counter + 1;
        setCounter( suma )
    }
    
    const handSubstract = () =>   setCounter( counter - 1 )

    const handleReset = () => {
        setCounter( value )
    } 

  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>

        {/* si no mando un EVENTO solo nombro la funcion */}
        <button onClick={ handleAdd }> +1 </button>
        <button onClick={ handSubstract }> -1 </button>
        <button onClick={ handleReset }> Reset </button>
    </>
  )
}

CounterApp.prototype = {
    value: PropTypes.number.isRequired
}

export { CounterApp }