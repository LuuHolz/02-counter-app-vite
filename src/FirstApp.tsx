// import PropTypes from 'prop-types';


// si las funciones o variables no tienen dependencia con lo utilizado dentro de FirstApp
//mejor dejarlo por fuera para que no ocupe espacio en la memoria el repintado
const newMessage = { 
    user: 'Lu',
    saludo: 'Saludar en H1'
};

const getMessage = () => {
    return newMessage.saludo
}

//-------------------------------------
// hay que desestructurar las props y extraer lo que vayamos a usar {}
                            // valores por defecto, ej abajo: defaultProps
const FirstApp = ( { title, subTitle}: { title: string, subTitle: string}) => {

    
    // if ( !title ) {
    //     throw new Error('No existe');
    // }

  return (
    <>
    {/* en vez de props.title con la desestructuracion: */}
        <h1 data-testid="test-title">{ title }</h1>
        <p>{ subTitle }</p>
        <p>{ subTitle }</p>


        <h2>{ getMessage() }</h2>

        <h2>{ newMessage.user}</h2>

        {/* Si necesito enviar el objeto entero:
      <code>{ JSON.stringify(newMessage) }</code> */}

    </>
  );
};

// FirstApp.propTypes = {
//     //la propiedad debe ser string y debe ser proporcionado
//     title: PropTypes.string.isRequired,
//     subTitle: PropTypes.string.isRequired,
// }

// FirstApp.defaultProps = {
//     subTitle: 'subTitle por default'
// }

export { FirstApp} ;
