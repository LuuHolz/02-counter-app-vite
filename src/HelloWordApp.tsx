
const HelloWordApp = () => {
  return (
    <h1>Hello Word App</h1>
  );
}

export  { HelloWordApp }

// FORMA RESUMIDA PORQUE SOLO DEVUELVE UNA COSA EL RETURN

// export const HelloWordApp = () => (<h1>Hello Word App</h1>);