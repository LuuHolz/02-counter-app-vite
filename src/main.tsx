import React from 'react'
import ReactDOM from 'react-dom/client';
import { HelloWordApp } from './HelloWordApp';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';
import './Styles.scss';


const App = () => {
    return (<h1> Hola Mundo </h1>);
}

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={ 5 }/>
        {/* <FirstApp title='Hola, Luciana' subTitle='subtitulo'/> */}
    </React.StrictMode>
)