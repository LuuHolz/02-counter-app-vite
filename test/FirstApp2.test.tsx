
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en FirstApp', () => {

   //variables reutilizables
   const title = 'Titutlo de prueba';
   const subTitle = 'SubTitulo de prueba';

   test('Debe de hacer match con le snapshot', () => {
     
        const { container } = render( <FirstApp title={title} subTitle={subTitle}/> )
        expect( container ).toMatchSnapshot();
   })

   test('Debe de mostrar el mensaje de title = Titutlo de prueba', () => {

        render( <FirstApp title={title} subTitle={subTitle}/> )

        //SCREEN: es el objeto que estamos renderizando. GETBYTEXT: que lo busquemos por el texto. TOBETRUTHY: tiene que existir
        expect( screen.getByText(title) ).toBeTruthy()
        
        //para ver el objeto
        // screen.debug();
   })
   
   test('Debe de mostrar el titulo en un H1', () => {

        render( <FirstApp title={title} subTitle={subTitle}/> )
                        //getByRole: buscar por algun objeo HTML
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
   })

   test('Debe de mostrar el subtitulo enviado por props', () => {

        render( <FirstApp title={title} subTitle={subTitle}/> )
        expect( screen.getAllByText(subTitle).length ).toBe( 2 );
})
   
    
    
});