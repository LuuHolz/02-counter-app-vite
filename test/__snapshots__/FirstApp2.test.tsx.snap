// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con le snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Titutlo de prueba
  </h1>
  <p>
    SubTitulo de prueba
  </p>
  <p>
    SubTitulo de prueba
  </p>
  <h2>
    Saludar en H1
  </h2>
  <h2>
    Lu
  </h2>
</div>
`;
